cloudify_agent/api/pm/base.py

```diff
@@ -217,10 +226,14 @@ def __init__(self, logger=None, **params):
             'name') or self._get_name_from_manager()
         self.queue = params.get(
             'queue') or self._get_queue_from_manager()
-        self.broker_url = params.get(
-            'broker_url') or defaults.BROKER_URL.format(
-            self.broker_ip,
-            self.broker_port)
+        # This is not retrieved by param as an option any more as it can make
+        # the _delete_amqp_queues function behave badly
```

Comment:
If you pass the pre-created broker_url to _delete_amqp_queues, it's used by something which, at the back-end, chokes if passed something that isn't a DNS name/IP without port (you have to pass the port in separately).

Since accepting the pre-formatted URL causes breakages in situations where a username, password, and/or port are supplied, I decided it was more sensible to disallow it rather than to support tokenising it.